jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -100,75 +100,96 @@ YUI.add('user-profile', function() {
           switchModel={props.switchModel}
           user={props.user} />
       );
+      const lists = [
+        <juju.components.UserProfileModelList
+          key='modelList'
+          ref='modelList'
+          addNotification={props.addNotification}
+          canCreateNew={props.canCreateNew}
+          controllerAPI={props.controllerAPI}
+          currentModel={props.currentModel}
+          hideConnectingMask={props.hideConnectingMask}
+          listModels={props.listModels}
+          showConnectingMask={props.showConnectingMask}
+          switchModel={props.switchModel}
+          user={props.user}
+          users={props.users} />,
+        <juju.components.UserProfileEntityList
+          key='bundleList'
+          ref='bundleList'
+          changeState={props.changeState}
+          charmstore={props.charmstore}
+          getDiagramURL={props.getDiagramURL}
+          type='bundle'
+          user={props.user}
+          users={props.users} />,
+        <juju.components.UserProfileEntityList
+          key='charmList'
+          ref='charmList'
+          changeState={props.changeState}
+          charmstore={props.charmstore}
+          getDiagramURL={props.getDiagramURL}
+          type='charm'
+          user={props.user}
+          users={props.users} />,
+        <juju.components.UserProfileAgreementList
+          key='agreementList'
+          ref='agreementList'
+          getAgreements={props.getAgreements}
+          user={props.user} />,
+        <juju.components.UserProfileBudgetList
+          key='budgetList'
+          ref='budgetList'
+          listBudgets={props.listBudgets}
+          user={props.user} />
+      ];
+      const toRender = [
+        'modelList',
+        'agreementList'
+      ];
+      if (window.flags && window.flags.blues) {
+        toRender.push('budgetList');
+      }
+      const charmstoreUser = props.users.charmstore;
+      const authenticated = charmstoreUser && charmstoreUser.user;
+      if (!props.interactiveLogin || authenticated) {
+        toRender.push('bundleList');
+        toRender.push('charmList');
+      }
+      const componentsToRender = lists.filter(list => {
```

Comment:
At first I was going to recommend doing this:
```JavaScript
const componentsToRender = lists.filter(
  list => toRender.indexOf(list.key) >= 0);
```

but then I realized that the => and >= are quite confusing.